great_tables/_utils_selenium.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+from types import TracebackType
+from typing import Literal
+from typing_extensions import TypeAlias
+from selenium import webdriver
+
+# Create a list of all selenium webdrivers
+WebDrivers: TypeAlias = Literal[
+    "chrome",
+    "firefox",
+    "safari",
+    "edge",
+]
+
+
+class _BaseWebDriver:
+
+    def __init__(self):
+        self.add_arguments()
+
+    def add_arguments(self): ...
+
+    def __enter__(self) -> WebDrivers | webdriver.Remote:
+        return self.driver
+
+    def __exit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc_value: BaseException | None,
+        traceback: TracebackType | None,
+    ) -> bool | None:
+        self.driver.quit()
+
+
+class _ChromeWebDriver(_BaseWebDriver):
+    def __init__(self, debug_port: int | None = None):
+        self.debug_port = debug_port
+        self.wd_options = webdriver.ChromeOptions()
+        super().__init__()
+        self.driver = webdriver.Chrome(self.wd_options)
```

Comment:
Could we consolidate the driver initialization across all these subclasses, by adding `cls_wd_options=ChromeOptions`, and `cls_driver=Chrome` as class attributes?

That way, the BaseClass could essentially require subclasses to specify those pieces. Then, the init would not have to be overridden by child classes.

This makes the child classes mostly about:

* specifying Driver and Option classes (as class attributes)
* customizing add_arguments method

Reply:
@machow , thanks for the quick response.  

Using `cls_driver` seems like a good idea, but I’m concerned that `cls_wd_options` might lead to issues. For example, what if we need to create two `Chrome` instances with different `ChromeOptions` in the future?

Reply:
@machow , I think I understand now. Does the latest commit align with what you had in mind?

Reply:
Yeah, thanks -- that looks really nice!